### PR TITLE
Add drives output & user filter

### PR DESCRIPTION
**Pull request by dragonchaser**

### CODE REVIEW

File: services/graph/pkg/service/v0/users.go [hunk 1]
```diff
@@ -171,6 +173,35 @@ func (g Graph) GetUser(w http.ResponseWriter, r *http.Request) {
...
+		wdu, err := url.Parse(g.config.Spaces.WebDavBase + g.config.Spaces.WebDavPath)
+		// TODO: filter this request for the user
+		f := listStorageSpacesUserFilter(user.GetId())
+		lspr, err := g.gatewayClient.ListStorageSpaces(r.Context(), &storageprovider.ListStorageSpacesRequest{
+			Opaque:  nil,
+			Filters: []*storageprovider.ListStorageSpacesRequest_Filter{f},
+		})
+		if err != nil {
+			g.logger.Err(err).Interface("query", r.URL.Query()).Msg("error getting storages")
+		}
```
Comment (micbar) [inline services/graph/pkg/service/v0/users.go:188]: we always need to check not only protocol errors but also Status code on the response.

Comment (dragonchaser) [reply services/graph/pkg/service/v0/users.go:198]: We can always return an user without a drives object if the return code from grpc is `!Code_CODE_OK`

**Comment by dragonchaser** [comment]
Reverted back to draft, there is a bug, only admins can list drives atm. Needs more logic in reva.

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/ocis/13901/42/8
```
runsh: Total unexpected failed scenarios throughout the test run:
apiSpaces/restoreSpaces.feature:63
apiSpaces/restoreSpaces.feature:64
```

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/ocis/13913/42/8

The same scenarios failed again:
```
runsh: Total unexpected failed scenarios throughout the test run:
apiSpaces/restoreSpaces.feature:63
apiSpaces/restoreSpaces.feature:64
```

It seems that either there is a regression (code to be fixed), or a change in behavior (tests to be adjusted)

**Comment by dragonchaser** [comment]
@phil-davis seems to be a bug in the code we introduced to reva: https://github.com/cs3org/reva/pull/3109

**Comment by dragonchaser** [comment]
will be superseded by #4244 , delete as soon as it is merged!

**Review by micbar — APPROVED** [review]
LGTM

**Pull request merged**